core/Pos/Core/Common/Types.hs

```diff
@@ -299,13 +299,20 @@ instance Bounded Coin where
 maxCoinVal :: Word64
 maxCoinVal = 45000000000000000
 
--- | Make Coin from Word64.
+-- | Just use the 'Coin' constructor. This is legacy but still used. Formerly
+-- it would fail with 'error' if the 'Word64' exceeds 'maxCoinVal', but now you
+-- have to 'checkCoin' if you care about that.
 mkCoin :: Word64 -> Coin
-mkCoin c
-    | c <= maxCoinVal = Coin c
-    | otherwise       = error $ "mkCoin: " <> show c <> " is too large"
+mkCoin c = either error (const coin) (runExceptT (checkCoin coin))
```

Comment:
It looks wrong to me. Why do you use `runExceptT` at all? `Either Text` is instance of `MonadError Text`.
`runExceptT (checkCoin coin)` has type `Either Text (Either Text Coin)` if I understand correctly.
And when `checkCoin` fails, `Left` will be in the inner `Either`. So `error` will never be called.
If you delete `runExceptT` completely, it should be fixed.

Reply:
> runExceptT (checkCoin coin) has type Either Text (Either Text Coin) if I understand correctly.

I think it's `Monad m => m (Either Text ())`. But right, the `m` will be specialized to `Either`. Will update.